### PR TITLE
Set Content-Type correctly with body::Form

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: test/client.jl [hunk 1]
```diff
@@ -242,4 +242,22 @@ end
         test(HTTP.delete(uri, headers; body=body, query=query), "DELETE")
         test(HTTP.delete(uri, headers, body; query=query), "DELETE")
     end
+    # Setting of Content-Type with body::Form
```
Comment (mattBrzezinski) [inline test/client.jl]: Can this be in its own `@testset`?

File: test/client.jl [hunk 2]
```diff
@@ -242,4 +242,22 @@ end
...
+    body = HTTP.Form(Dict())
+    uri = "https://httpbin.org/post"
+    for r in (HTTP.request("POST", uri, headers, body), HTTP.post(uri, headers, body))
+        @test r.status == 200
+        json = JSON.parse(IOBuffer(HTTP.payload(r)))
+        @test startswith(json["headers"]["Content-Type"], "multipart/form-data; boundary=")
+    end
+    # Deprecation of HTTP.post without header for body::Form
```
Comment (mattBrzezinski) [inline test/client.jl]: This should probably be moved into its own `@testset` as well.

File: test/client.jl [hunk 3]
```diff
@@ -242,4 +242,22 @@ end
...
+    proj = normpath(joinpath(pathof(HTTP), "..", "..", "Project.toml"))
+    vers = VersionNumber(match(r"^version\s*=\s*\"(.*?)\"$"m, read(proj, String)).captures[1])
```
Comment (mattBrzezinski) [inline test/client.jl]: Can there be a comment explaining the `regex`? Makes it easier to know what you're matching at a quick glance.

File: src/multipart.jl [hunk 4]
```diff
@@ -125,9 +125,6 @@ end
 content_type(f::Form) = "Content-Type" =>
                         "multipart/form-data; boundary=$(f.boundary)"
 
+# To be deprecated in HTTP@0.9
```
Comment (mattBrzezinski) [inline src/multipart.jl]: Can we log an issue for this so it's not forgotten?

File: test/client.jl (hunk 1 above)
Comment (fredrikekre) [reply test/client.jl]: I guess.

File: src/multipart.jl (hunk 4 above)
Comment (fredrikekre) [reply src/multipart.jl]: Thats why I conditioned the tests on the version; when the version is bumped CI will fail and then someone has to enable this deprecation.

**Comment by mattBrzezinski** [comment]
I don't have approve permissions. But looks good to me, ship it!

**Comment by ararslan** [comment]
Travis is consistently timing out running the server tests, but the tests here all run before that and passed, so I think we're good.

**Pull request merged**